pypdf/_xobj_image_helpers.py

```diff
@@ -125,6 +125,20 @@ def _get_imagemode(
     return mode, mode == "CMYK"
 
 
+def _extended_image_frombytes(
+    mode: str, size: Tuple[int, int], data: bytes
+) -> Any:  # Any = ImageType
+    try:
+        img = Image.frombytes(mode, size, data)
+    except ValueError as exc:
+        if len(data) % (size[0] * size[1]) != 0:
+            raise exc
+        k = size[0] * size[1] * len(mode) / len(data)
```

Comment:
Avoid calculating the pixel count twice here by assigning it to a variable. Should increase the readability as well.